REMIND/EDGE-T variable harmonization

johannah-pik opened the pull request:
**Harmonize data to last REMIND iteration in coupled mode**
see issue https://github.com/pik-piam/transportIssues/issues/92#issue-3948095918

related remind PR: https://github.com/remindmodel/remind/pull/2320

_Approach:_
1. **Check status before the harmonization:** 
New function checkConvergence has been introduced that is called before reportEdgeTransport(). It checks and compares the parameters in the fulldata.gdx from the last REMIND iteration with the same parameter edget would theoretically report back to REMIND from the last edget iteration. trackConvergence.csv is saved in the EDGE-T folder of a REMIND run. It documents the original deviations before the harmonization for all outliers > 0.5% of the EDGETvalue.
For further details trackREMINDvsEDGETparameterEnergyIntensity.csv and trackREMINDvsEDGETparameterEnergyService.csv are provided for all regions and timesteps. 
2. **Harmonize energy service demand:** 
_The energy Service demand on REMIND CES leave level (pass_sm, freight_sm, pass_long, freight_long) is read in again from the last REMIND run and the energy service demand on fleet level in edget is rescaled to match it. Every other input is kept as it was before. 
As the reporting of FE, emissions and total cost variables depends on the ES demands, this then automatically rescales the edget-reported FE, emissions and total cost with the same ratio of (final REMIND run ES) : (final edget run ES)_
- Switch _isHarmonized_ has been introduced to reportEdgeTransport(). It is turned off by default and only enabled using reportEdgeTransport() to attach transport variables to a REMIND.mif 
- The energy Service demand on sector level is read in again from the last REMIND run and the energy service demand on fleet level in edget is rescaled to match it. Everything else is kept as it is. 
- The energy intensity is not an endogenous parameter in REMIND and does not change over REMIND iterations -> it is expected, that rescaling the energy service demand in edgeTransport is sufficient to solve all summation problems
- Rescaling the energy service demand on fleet level without running edget again leads to inconsitencies e.g. for annual mileage/stock/energy service demand. Hence this postprocessing step should ideally be only used to remove small remaining differences between both models after convergence. That is why further analysing model convergence in the future is essential.
4. **Compare variables that are reported by remind2 and by reportEdgeTransport for the REMIND.mif file after harmonization**
- A mapping has been introduced for the shared variables  helpersCommonVarsEDGETremind.csv
- Both variables are compared and outliers above 0.5% of REMINDvalue are stored in checkREMINDvsEDGETmifVariables.csv in the EDGE-T folder 
5. **Filter out shared variables** on edget side + those that have no remind2 equivalent but the same name as another remind2 variable -> No duplicates in reporting

_Open issues_
Tests have been performed and revealed that the general approach works well. The energy service demand deviations are harmonized and also resolve the issues for FE and Emi. But for certain timesteps the energy Intensity between edget and REMIND differs. That should not be the case and is a bug that is unrelated to the assumed convergence problems (as explained above).
**Details:**
1. /p/projects/remind/runs/REMIND_2026_03_18/remind/output/SSP2-NPi_2026-03-18_18.14.07
- check parameter plausibility using checkConvergence(): The energy intensity has outliers in 2005 and 2010 that deviate more than 1% from the edget value -> This is most likely a problem of deviation in historical timesteps between different edget policy scenarios while REMIND is fixed on Mix1 until 2010. The deviations in edgeTransport standalone standard scenarios were revealed using the following test script [testIdentityHistoricalTimestepsStandardScenario.txt](https://github.com/user-attachments/files/26250624/testIdentityHistoricalTimestepsStandardScenario.txt)
- check final variables in REMIND mif:
As expected from the energy intensity parameter check, FE and emission values deviate for 2005 and 2010, all other timesteps are fine

2. /p/projects/remind/runs/REMIND_2026_03_18/remind/output/SSP2-PkBudg650_2026-03-18_19.41.06
_climate policy run with cm_startyear = 2030_
- check parameter plausibility using checkConvergence(); the energy intensity has outliers until 2025 that deviate more than 1% of edget value
- check final variables in REMIND mif: As expected from the energy intensity parameter check, FE and emission values deviate for years < 2030, all other timesteps are fine

_Current  problem analysis:_
- The deviation in historical timesteps for different policy scenarios in edget standalone is a bug. REMIND full.lst shows, that REMIND parameters like pm_fe2es are filled with Mix1 values first and are then only updated after 2010. That explains well the issues in SSP2-NPi_2026-03-18_18.14.07 (cm_startyear 2005).
- The issues with SSP2-PkBudg650_2026-03-18_19.41.06 (cm_startyear 2030, not only 2005 and 2010 deviate as for the startyear = 2005 run, but also 2015, 2020 and 2025) could be an indication that the parameter fixing for cm_startyear is affected as well (maybe as a consequence of the above described bug with the standalone policy scenarios)

johannah-pik committed:
varHarm

johannah-pik committed:
remove duplicates

johannah-pik committed:
fix storeData call

johannah-pik committed:
buildLib

johannah-pik committed:
remove duplicate variable renaming

johannah-pik committed:
buildlib

johannah-pik committed:
bugfix in varHarmMap + tests

johannah-pik committed:
variable harmonization

johannah-pik committed:
Update harmonizeOutput.R

johannah-pik committed:
updates

johannah-pik committed:
Merge branch 'master' of github.com:pik-piam/reporttransport into VarHarm

johannah-pik committed:
update

johannah-pik committed:
update varharm

johannah-pik committed:
Actuualize var Harm

johannah-pik committed:
Delete harmonizeOutput.R

This script was the first attempt for variable harmonization

johannah-pik committed:
buildlib

johannah-pik committed:
Update README.md

johannah-pik committed:
update reporting

johannah-pik committed:
Update commonVarsEDGETremind.csv

johannah-pik committed:
Update reportEdgeTransport.R

johannah-pik committed:
Update commonVarsEDGETremind.csv

johannah-pik committed:
get gdxPath from data

johannah-pik committed:
Merge branch 'master' of github.com:pik-piam/reporttransport into VarHarm

johannah-pik committed:
Revert "get gdxPath from data"

This reverts commit 2eb6d06fdfd2b71a02c89b6ff5306c8a579104ed.

johannah-pik committed:
draft better solution

johannah-pik committed:
Merge branch 'master' of github.com:pik-piam/reporttransport into VarHarm

johannah-pik committed:
fix varHarm

johannah-pik committed:
fixes

johannah-pik committed:
testing

johannah-pik committed:
adjust filename

robertpietzcker commented:
wow, impressive work, @johannah-pik !!

robertpietzcker commented:
When trying to understand the overall logic, I got a few questions / comments :-)

1. on your point "2. Harmonize energy service demand:"
> The energy Service demand on sector level is read in again from the last REMIND run and the energy service demand on fleet level in edget is rescaled to match it. Everything else is kept as it is.

is only ES rescaled, or also FE and Emi? 

if only ES is rescaled, then the detailed FE reported by EDGE-T would still be different than the aggregate FE reported from REMIND?

robertpietzcker commented:
on "3. Compare variables that are reported by remind2 and by reportEdgeTransport for the REMIND.mif file after harmonization"

is the mapping file ONLY used to compare the values between REMIND and EDGE-T, or is it ALSO used to rename the EDGE-T variables when added to the REMIND reporting?

robertpietzcker commented:
on "4. Filter out shared variables on edget side + those that have no remind2 equivalent but the same name as another remind2 variable -> No duplicates in reporting"

a) I don't understand the formulation "Filter out shared variables on edget side + those that have no remind2 equivalent but the same name as another remind2 variable" 
what is meant with "shared variables on edget side"? 
b) on "those that have no remind2 equivalent but the same name as another remind2 variable" - this refers to the problem that EDGE T has bunkers outside of normal transport reportings, while REMIND includes it? 

c) a comment: I think it would be important to have at least the top-level FE value (I guess "FE|Transport with bunkers"?) 
still contained in the REMIND-MIF WITHOUT harmonization - so to have a variable in the REMIND-mif called "FE|Transport with bunkers unharmonized from EDGE-T" (or some other name with similar information content).

the reason for this is that only the mif is regularly looked at. The other files are hugely important for understanding where the problems come from, but they will not be normally looked at - so to notice in the future that either a new problem has arisen or the old one is still there, we need such a low-barrier information in the mif.

johannah-pik commented on R/reportEdgeTransport.R:
```diff
@@ -211,6 +227,50 @@ reportEdgeTransport <- function(folderPath = file.path(".", "EDGE-T"), data = NU
+        # Variables reported by edgeTRansport that should be removed in the REMIND mif to avoid confusion/duplicates
+        # 1. Remove duplicates that are reported by remind2 already + 2. variables that do not have an remind2 equivalent
+        # but follow a different naming convention (e.g. FE|Transport|Pass includes bunkers in remind2 and not in reporttransport)
+        varsToBeRemoved <- c(unique(remindEDGEvarMap$variable),
```
here the variables are removed

johannah-pik commented:
Thanks for the questions @robertpietzcker!

1. No- I am rescaling the energy service demand at the beginning of the reporting. Hence all upstream variables are calculated based on the harmonized energy service demand. As the energy intensity (in theory) do not vary between REMIND iterations and the emission factors are anyway used from the fulldata.gx of the last REMIND iteration, this fixes deviations for FE and Emissions as well.  helpersCommonVarsEDGETremind.csv includes and compares these variables as well, and it works :) (without the prevailing bug of historical variations we do not want)
2. The mapping is only used to compare the variables. As the differences in the end are numerical (in theory, without the bug), all edget variables (that are also reported in remind2) can then simply be removed from the REMIND.mif. + additional variables c("FE|Transport|Pass|Liquids", "FE|Transport|Pass", "FE|Transport|Freight", "Emi|CO2|Energy|Demand|Transport|Liquids", "Emi|CO2|Energy|Demand|Transport|Gases"). These variables cannot be mapped directly on each other but would lead to duplicates or confusion. 
In general this is due to the bunker/no bunkers variable naming difference. So "FE|Transport|Pass" in edgeTransport would be mapped to the a REMIND equivalent like this: "FE|Transport|Pass w/o bunkers". But this variable is simply not reported by remind2. So it cannot be mapped. Still keeping "FE|Transport|Pass" from edget in the REMIND.mif would create conflicts with the REMIND variable that includes bunkers but has the same name.

The transport.mif to perform a detailed transport sector analysis is still produced with the unharmonized edget data only. This makes sense as e.g. the consistency of capital cost/stock variables vs energy service demand is damaged by the harmonization. So I prefer to keep a consitent data set for our detailed transport stuff. Also no RDS files in the EDGE-T folder are overwritten. So only the edgeTransport variables attached to the REMIND.mif are affected. So all data reported for the coupled REMIND/EDGE-T system is harmonized and should pass summation checks now (in theory haha)

johannah-pik commented:
On your last point I disagree. We have a very detailed and easy to access way to look at the problem in the future. We even report directly the relative deviations and a final check.   
I think for people that are just using the REMIND mif to report the result to projects we should include as little ambiguity as possible. We found all sorts of variable usage in the past.. So I think whatever enters the REMIND mif should be harmonized and unique

robertpietzcker commented:
> On your last point I disagree. We have a very detailed and easy to access way to look at the problem in the future. We even report directly the relative deviations and a final check.

I would also be fine if the deviation is reported in the MIF. it just needs to be in the MIF - that is the main thing necessary to keep the barrier low enough to have a slight chance of this getting seen. 

I have a very strong opinion on this, based on my last decade working with REMIND and mifs. 

> I think for people that are just using the REMIND mif to report the result to projects we should include as little ambiguity as possible. We found all sorts of variable usage in the past.. So I think whatever enters the REMIND mif should be harmonized and unique

yes, the MIF should be as unambiguous as possible. 

Still, it is the one place to present the largets set of information about the model state and results. Having a variable with a very clear name that screams "don't use me for project reporting" is totally fine and will not cause any confusion. 

actually, given your comment, what about writing a variable "FE|Transport|ratio of EDGE-T to REMIND before harmonization" to the MIF - it is impossible to mistake that as a real output variable.

robertpietzcker commented:
> Both variables are compared and outliers above 1% of REMINDvalue are stored in checkREMINDvsEDGETmifVariables.csv in the EDGE-T folder

is there a specific reason for choosing 1%? 

my hope would be that if remaining differences are just numerics from the various calculations, the filter could be set at 0.1% or even 0.01% without getting many more variables reported? 

I would just feel a bit safer with knowing "they are identical to 0.1%", compared to "identical to 1%".

johannah-pik commented:
Could you may describe a use case, where you think this would help in reality?
My personal view and experience on that is that we have so many variables and also plots in the compscen, that are partially broken and simply ignored that I would not bet on this in any case and would not want to spent time on implementing that. 

What I do have trust in is, that together with RSE we will look into fixing the convergence issues of edget and from now on we will collect usable and easy to process data to do so for all future runs. And when the other bug is fixed and we have a better understanding of the problem, we could implement a test, that throws an error, when the models do not converge

robertpietzcker commented:
> Could you may describe a use case, where you think this would help in reality?

yes. for each validation, each run I do in a project, I am looking at the mif. either directly on the cluster, or in excel. 
when selecting  the variable from the drop-down menu, the additional cost of also clicking on the "FE|Transport|ratio of EDGE-T to REMIND before harmonization" and checking that when anyway looking at some transport-FE thing is ~2-5 sec. 

if the information is in a separate file, I will have forgotten the file name in 2 weeks, and simply will not look into it regulary. 

Only when I see a discrepancy in the variable above will I think about searching through stuff and looking it up. 


your questions actually help a lot improving the proposal by forcing me to visualize the process: 
I would now go for the name ""FE|Transport|a ratio of EDGE-T to REMIND before harmonization in percent"

with the "a" it will always be at the top of the list :-)

johannah-pik commented:
hmm but what will you do with this information? I think without context it is kind of useless

robertpietzcker commented:
> hmm but what will you do with this information? I think without context it is kind of useless

it tells me "do I need to worry about this/ look deeper into this"

johannah-pik commented:
> > hmm but what will you do with this information? I think without context it is kind of useless
> 
> it tells me "do I need to worry about this/ look deeper into this"

I think the sad answer is yes.

So this whole fix is about masking something that is actually a problem. So I fear we are not there yet to say "this parameter is fine like that and you should start worrying when it changes". As soon as we have capacities, we should start looking deeper into this and implement a long-term fix/test

robertpietzcker commented:
right - and then having this variable in the mif helps for 
a) keeping it on my it radar, and 
b) giving me a better understanding for specific run if the harmonization removed a 5% difference or a 95% difference

johannah-pik commented:
> right - and then having this variable in the mif helps for a) keeping it on my it radar, and b) giving me a better understanding for specific run if the harmonization removed a 5% difference or a 95% difference

I can implement it in another PR on my next model maintenance day

robertpietzcker commented:
I think I now better understand the "filter out variables" part. 
maybe changing the description up above in the PR to 
```
Filter out shared variables
1. when a certain content is reported in a variable both from EDGE-T and from REMIND (as defined in  helpersCommonVarsEDGETremind.csv), the EDGE-T variable is now removed from the EDGE-T reporting before it is attached to the REMIND.mif. 

2. Variables from EDGE-T that have the same name as a remind2 variable, but contain something different. This arises because in EDGE-T, the scope of "Transport" is defined as "w/o bunkers", while in REMIND it is defined as "w/ bunkers"

```

might make it a bit clearer?

robertpietzcker commented:
> I can implement it in another PR on my next model maintenance day

thanks a lot!

robertpietzcker commented:
> No- I am rescaling the energy service demand at the beginning of the reporting. Hence all upstream variables are calculated based on the harmonized energy service demand. As the energy intensity (in theory) do not vary between REMIND iterations and the emission factors are anyway used from the fulldata.gx of the last REMIND iteration, this fixes deviations for FE and Emissions as well. helpersCommonVarsEDGETremind.csv includes and compares these variables as well, and it works :) (without the prevailing bug of historical variations we do not want)


to make this clearer, what about rephrasing
`The energy Service demand on sector level is read in again from the last REMIND run and the energy service demand on fleet level in edget is rescaled to match it. Everything else is kept as it is.`

to something like 

```
The energy Service demand on REMIND CES leave level (pass_sm, freight_sm, pass_long, freight_long) is read in again from the last REMIND run and the energy service demand on fleet level in edget is rescaled to match it. Every other input is kept as it was before. 
As the reporting of all extensive variables depends on the ES demands, this then automatically rescales the edget-reported FE, emissions and total cost with the same ratio of (final REMIND run ES) : (final edget run ES)
```
?

johannah-pik committed:
include review suggestions

johannah-pik committed:
Update reportEdgeTransport.R

ahagen-pik commented:
Thank you @johannah-pik and also @robertpietzcker for the nice discussion and help refining the documentation.

I am completely fine with the solution atm and also with adding a variable "FE|Transport|a ratio of EDGE-T to REMIND before harmonization in percent". [Maybe small suggestion to use something like "FE|Transport|a - ratio of EDGE-T to REMIND before harmonization in percent" if we want to keep the 'a' in the name, otherwise it sounds a bit like there were different ratios.]

Regarding your very last point in the description, @johannah-pik:

> The issues with SSP2-PkBudg650_2026-03-18_19.41.06 (cm_startyear 2030) could be an indication that the parameter fixing for cm_startyear is affected as well (maybe as a consequence of the above described bug with the standalone policy scenarios)

Just a reminder that we have no 'fixing' in EDGE-T. What we do with cm_startyear in EDGE-T is that we change the scenario assumptions with startyear, such that before startyear the assumptions are the same as for NPi2025 (at least in the default usage case). So having the problem in "SSP2-PkBudg650_2026-03-18_19.41.06" as well, rather shows that the startyear fixing works and thus, like you suggested, the bug we have for a simple NPi run - that REMIND used Mix1 values but in EDGET the Mixes _incorrectly_ lead to differences in historic time steps - carries over. 
Maybe this is what you meant already, just wanted to clarify. So fixing the bug will probably be sufficient for all scenarios.

johannah-pik commented:
Thanks for your review @ahagen-pik.
I will add the "-" to the variable name. 

> The issues with SSP2-PkBudg650_2026-03-18_19.41.06 (cm_startyear 2030) could be an indication that the parameter fixing for cm_startyear is affected as well (maybe as a consequence of the above described bug with the standalone policy scenarios)

With my last point here I just wanted to point out that I noticed that for the startyear = 2030 run not only 2005 and 2010 deviate (as for the startyear = 2005 run), but also 2015, 2020 and 2025..

johannah-pik committed:
Update reportEdgeTransport.R

johannah-pik committed:
bugfix

johannah-pik committed:
bugdix for 21 regional resolution

johannah-pik committed:
bugfix missing downstream variable

johannah-pik committed:
Update reportEdgeTransport.R

johannah-pik committed:
bugfix

its not a problem, when the harmonization runs twice (incoming variables are labelled as REMINDvars and there is now version mix up)

johannah-pik committed:
Adjust filter tolerance

johannah-pik commented:
I performed more tests and fixed some remaining bugs.
fyi: @robertpietzcker for some reason excel is not able (for some cases) to interprete the numerical data from the mif file correctly.
<img width="1212" height="990" alt="image" src="https://github.com/user-attachments/assets/f5edfa43-109f-48ad-85f9-72f85ce7a7b6" />
According to chatgpt this is a common problem with excel (I checked in R and by opening the mif with a basic editor, the problem only exists when you open the mif file with excel)

johannah-pik committed:
deviation in percent

johannah-pik commented:
Furthermore I changed the sensitivity of the filter back to 0.5% (instead of 0.1%) as more testing revealed that we catch many numerical deviations otherwise (e.g. the gases variables in late years)

johannah-pik commented:
For this variable "FE|Transport|a ratio of EDGE-T to REMIND before harmonization in percent" I realized unfortunately that my prior implementation reported the ratio after the harmonization. From the structure of reporting.R in remind, it is not obvious where to get this data from: The REMIND mif file is written inside convGDX2MIF(). reportEdgeTransport() harmonizes the data before calculating FE (for very good reasons explained above). Comparing the harmonized data as a check to the data stored in the REMIND mif file is important to check whether harmonization was succesfull, and where not (other bugs). 

So we load the REMIND mif file once in reportEdgeTRansport(). And now I did not see another quick solution to calculate "FE|Transport|a ratio of EDGE-T to REMIND before harmonization in percent" without loading it again in reporting.R and comparing it to the unharmonized data calculated for the Transport.mif.

This is not very elegant and takes a lot of run time. 
Here it is: https://github.com/remindmodel/remind/pull/2320/changes

jmuessel commented on inst/helpersCommonVarsEDGETremind.csv:
```diff
@@ -0,0 +1,106 @@
+Emi|CO2|Energy|Demand|Transport|w/o bunkers;Emi|CO2|Energy|Demand|Transport
+Emi|CO2|Energy|Demand|Transport|International Bunkers;Emi|CO2|Energy|Demand|Transport|Bunkers
+Emi|CO2|Energy|Demand|Transport|w/ bunkers;Emi|CO2|Energy|Demand|Transport with bunkers
+Emi|CO2|Energy|Demand|Transport;NA
```
Is this w/o bunkers from the REMIND side? If yes, it does not follow the REMIND variable name logic, but we could add `Emi|CO2|Energy|Demand`. If not, it is a duplicate of the variable name in line 103, while 103 does not follow the REMIND variable name logic. In this case, I would rather put the NA in line 103 and `Emi|....|Transport with bunkers` into line 104.

johannah-pik replied:
> Hi @jmuessel, thanks for reviewing the mapping in detail! I think I did that back in the days because of the bunkers correction. When I remember it correctly, "Emi|CO2|Energy|Demand|Transport" in remind2 included bunkers only for "World" (@fschreyer can you help here?). Thas why I mapped our variables on the clear [w/o bunkers and w/ bunkers variables](https://github.com/pik-piam/remind2/blob/836ca894e24dc3a971be2afe66cecd404f10a6df/R/reportEmi.R#L4007-L4010). But it looks like the spelling is off - I will check that later on

jmuessel commented:
@johannah-pik, great work and thank you! Did you see my comment in `ins/helpersCommonVarsEDGETremind.csv`? I'm not sure if you see my question because it says "pending"

johannah-pik commented:
> @johannah-pik, great work and thank you! Did you see my comment in `ins/helpersCommonVarsEDGETremind.csv`? I'm not sure if you see my question because it says "pending"

aah no :( You need to click on "submit review". Otherwise the comments are only visible to you.. Maybe github changed that at one point.. I also got confused with it several times recently

robertpietzcker commented:
Thanks so much for implementing this fix, @johannah-pik !!

robertpietzcker commented:
> fyi: @robertpietzcker for some reason excel is not able (for some cases) to interprete the numerical data from the mif file correctly.

wow, that sucks (from excel side). 

If I am not mistaken, all energy system variables are written with max 7 digits after the decimal point. 
Would it be possible to do the same for the EDGE-T variables? I think that might suppress this excel error (at least I have never before come upon this) 

(not urgent)

@johannah-pik

johannah-pik commented:
> > fyi: @robertpietzcker for some reason excel is not able (for some cases) to interprete the numerical data from the mif file correctly.
> 
> wow, that sucks (from excel side).
> 
> If I am not mistaken, all energy system variables are written with max 7 digits after the decimal point. Would it be possible to do the same for the EDGE-T variables? I think that might suppress this excel error (at least I have never before come upon this)
> 
> (not urgent)
> 
> @johannah-pik

I will try this soon!

fbenke-pik commented on R/reportToREMINDfinalEnergyShares.R:
```diff
@@ -25,6 +24,7 @@ reportToREMINDfinalEnergyShares <- function(fleetFEdemand, timeResReporting, dem
   FEshares[, c("sumvalue", "maxtech") := NULL]#
-
+  test <- FEshares[, sumvalue := sum(value), by = c("tall", "all_regi", "all_in")]
+  if (nroww(test[!sumvalue==1] > 0)) stop("Something went wrong in reportToREMINDfinalEnergyShares()")
```
This is a bug

robertpietzcker commented:
Good catch, @jmuessel !
the classifiers "w/o Bunkers" and "w/ Bunkers" need to come right after `CO2` - so `Emi|CO2|w/o Bunkers|Energy|Demand|Transport` and `Emi|CO2|w/ Bunkers|Energy|Demand|Transport` instead of what is currently in the mapping

 @johannah-pik 

> When I remember it correctly, "Emi|CO2|Energy|Demand|Transport" in remind2 included bunkers only for "World"

yes, that was the case earlier, but we decided to abandon this confusing differentiation between World and regions completely:
https://github.com/pik-piam/remind2/pull/709

johannah-pik commented:
Nice! I did not remember the status on this. 
I will correct the spelling of the bunker/no bunkers variables next week.. 


> Good catch, @jmuessel ! the classifiers "w/o Bunkers" and "w/ Bunkers" need to come right after `CO2` - so `Emi|CO2|w/o Bunkers|Energy|Demand|Transport` and `Emi|CO2|w/ Bunkers|Energy|Demand|Transport` instead of what is currently in the mapping
> 
> @johannah-pik
> 
> > When I remember it correctly, "Emi|CO2|Energy|Demand|Transport" in remind2 included bunkers only for "World"
> 
> yes, that was the case earlier, but we decided to abandon this confusing differentiation between World and regions completely: [pik-piam/remind2#709](https://github.com/pik-piam/remind2/pull/709)
